Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -32,14 +32,50 @@ AZ_CVAR(
     AZ::ConsoleFunctorFlags::DontReplicate | AZ::ConsoleFunctorFlags::DontDuplicate,
     "If set, enables experimental Document Property Editor support, replacing the Reflected Property Editor where possible");
 
+
+template <class T>
+void DetachAndHide(T* widget)
+{
+    if (widget)
+    {
+        widget->hide();
+        widget->setParent(nullptr);
+    }
+}
+
 namespace AzToolsFramework
 {
-    DPELayout::DPELayout(int depth, QWidget* parentWidget)
-        : QHBoxLayout(parentWidget)
-        , m_depth(depth)
+    DPELayout::DPELayout(QWidget* parent)
+        : QHBoxLayout(parent)
+        , m_depth(-1)
     {
     }
 
+    void DPELayout::Init(int depth, [[maybe_unused]] QWidget* parentWidget)
+    {
+        m_depth = depth;
+    }
+
+    void DPELayout::Clear()
+    {
+        m_showExpander = false;
+        m_expanded = true;
+        delete m_expanderWidget;
+        m_expanderWidget = nullptr;
+        m_sharePriorColumn.clear();
+        m_cachedLayoutSize = QSize();
+        m_cachedMinLayoutSize = QSize();
+        for (auto index = count() - 1; index >= 0; --index)
+        {
+            auto theItem = takeAt(index);
+            auto theWidget = theItem->widget();
+            if (theWidget)
+            {
+                theWidget->deleteLater();
+            }
+        }
```

Comment:
FYI:"This only works when `count()` returns a signed type.

When an unsigned type is used, this would result in an infinite loop.
Luckily Qt `count()` function returns an `int`, but the code can be authored in a way that works for both signed and unsigned cases by deferring decrementing the index until it is used
```suggestion
        for (int index = count(); index > 0; --index)
        {
            auto theItem = takeAt(index - 1);
            auto theWidget = theItem->widget();
            if (theWidget)
            {
                theWidget->deleteLater();
            }
        }
```

Reply:
good catch! I'll take that change directly :)